database/migrations/20191105000804_createTables.js

```diff
@@ -0,0 +1,72 @@
+
+exports.up = function(knex) {
+  return knex.schema
+  
+  // users
+  .createTable("users", tbl => {
+    tbl.increments();
+    tbl.string('username').notNullable().unique();
+    tbl.string('password').notNullable();
+    tbl.json("name").notNullable();
```

Comment:
I’m a little reticent to make things that have known requirements JSON (we could decide to make name broken down to first and last both of type string). 

Extremely open to discussion, I just want to make sure that we are thinking about the quality and ease of our design. Will this decision require us to do more validation elsewhere? Or will it require the front end to handle more complex cases of conditional rendering?

Reply:
It was around the idea that there may be a potential for a user to be created without knowing the name. Like if the landlord is sending a request to a tenant to sign up. However, it may be more productive to add another table for landlords sending tenant requests. Let's discuss, there most likely is a better solution I'm not seeing.